wgengine/magicsock/magicsock.go

```diff
@@ -323,6 +322,17 @@ type Conn struct {
 	staticEndpoints views.Slice[netip.AddrPort]
 }
 
+// sendToBlockedWithEPERMWarning is set when a sendto call returns an error containing EPERM.
+var sendToBlockedWithEPERMWarning = health.Register(&health.Warnable{
+	Code:     "firewall-blocking-udp",
+	Severity: health.SeverityMedium,
+	Title:    "Tailscale blocked by system firewall",
+	Text: func(args health.Args) string {
+		return "The operating system firewall is blocking UDP sends, preventing direct connections. Try reconfiguring your firewall or checking the configuration of EDR software."
+	},
+	ImpactsConnectivity: true,
```

Comment:
Is it the case that this will always block Internet connectivity, or does it just force relayed connections? For context, `ImpactsConnectivity` changes the tray icon to an exclamation mark.

Reply:
yeah, forces relay, is there a half-way warning sign?

Reply:
We should build it! :)